diff --git a/admin/settings.php b/admin/settings.php
index fca9a3cd..b5fbcd27 100644
--- a/admin/settings.php
+++ b/admin/settings.php
@@ -105,6 +105,28 @@ $hostname = trim(file_get_contents("/etc/hostname"), "\x00..\x1F");
 ?>
 
 <?php
+// Lancache settings
+$LCServices = [];
+$LCServicesActive = [];
+
+//TODO: Read cache_domains.json
+if (file_exists("/etc/pihole-extra/cache-domains/cache_domains.json")) {
+    $cache_domains = json_decode(file_get_contents("/etc/pihole-extra/cache-domains/cache_domains.json"));
+    $cache_domains = $cache_domains->cache_domains;
+
+    foreach ($cache_domains as $key => $value) {
+        $LCServices[$value->name] = ['description' => $value->description];
+        if (isset($setupVars["LANCACHE_SERVICE_" . strtoupper($value->name)])
+            && $setupVars["LANCACHE_SERVICE_" . strtoupper($value->name)] == 1) {
+            $LCServices[$value->name]['enabled'] = true;
+        } else {
+            $LCServices[$value->name]['enabled'] = false;
+        }
+    }
+}
+?>
+
+<?php
 // DNS settings
 $DNSservers = [];
 $DNSactive = [];
@@ -224,7 +246,7 @@ if (isset($setupVars["API_PRIVACY_MODE"])) {
 ?>
 
 <?php
-if (isset($_GET['tab']) && in_array($_GET['tab'], array("sysadmin", "blocklists", "dns", "piholedhcp", "api", "privacy", "teleporter"))) {
+if (isset($_GET['tab']) && in_array($_GET['tab'], array("sysadmin", "blocklists", "dns", "piholedhcp", "lancache", "api", "privacy", "teleporter"))) {
     $tab = $_GET['tab'];
 } else {
     $tab = "sysadmin";
@@ -238,6 +260,7 @@ if (isset($_GET['tab']) && in_array($_GET['tab'], array("sysadmin", "blocklists"
                 <li<?php if($tab === "blocklists"){ ?> class="active"<?php } ?>><a data-toggle="tab" href="#blocklists">Blocklists</a></li>
                 <li<?php if($tab === "dns"){ ?> class="active"<?php } ?>><a data-toggle="tab" href="#dns">DNS</a></li>
                 <li<?php if($tab === "piholedhcp"){ ?> class="active"<?php } ?>><a data-toggle="tab" href="#piholedhcp">DHCP</a></li>
+                <li<?php if($tab === "lancache"){ ?> class="active"<?php } ?>><a data-toggle="tab" href="#lancache">Lancache</a></li>
                 <li<?php if($tab === "api"){ ?> class="active"<?php } ?>><a data-toggle="tab" href="#api">API / Web interface</a></li>
                 <li<?php if($tab === "privacy"){ ?> class="active"<?php } ?>><a data-toggle="tab" href="#privacy">Privacy</a></li>
                 <li<?php if($tab === "teleporter"){ ?> class="active"<?php } ?>><a data-toggle="tab" href="#teleporter">Teleporter</a></li>
@@ -901,8 +924,149 @@ if (isset($_GET['tab']) && in_array($_GET['tab'], array("sysadmin", "blocklists"
                             </div>
                         </div>
                     </form>
-                </div>
-                <!-- ######################################################### API and Web ######################################################### -->
+                </div>        
+				<!-- ######################################################### Lancache ######################################################### -->
+                <?php
+                // Lancache server
+                if (isset($setupVars["LANCACHE_ACTIVE"])) {
+                    if ($setupVars["LANCACHE_ACTIVE"] == 1) {
+                        $LANC = true;
+                    } else {
+                        $LANC = false;
+                    }
+                    // Read setings from config file
+                    /*if (isset($setupVars["LANCACHE_SERVICE_APPLE"]) && $setupVars["LANCACHE_SERVICE_APPLE"] == 1) {
+                        $LANC_SERVICE_APPLE = true;
+                    } else {
+                        $LANC_SERVICE_APPLE = false;
+                    }
+
+
+                    if (isset($setupVars["DHCP_START"])) {
+                        $DHCPstart = $setupVars["DHCP_START"];
+                    } else {
+                        $DHCPstart = "";
+                    }
+                    if (isset($setupVars["DHCP_END"])) {
+                        $DHCPend = $setupVars["DHCP_END"];
+                    } else {
+                        $DHCPend = "";
+                    }
+                    if (isset($setupVars["DHCP_ROUTER"])) {
+                        $DHCProuter = $setupVars["DHCP_ROUTER"];
+                    } else {
+                        $DHCProuter = "";
+                    }
+
+                    // This setting has been added later, we have to check if it exists
+                    if (isset($setupVars["DHCP_LEASETIME"])) {
+                        $DHCPleasetime = $setupVars["DHCP_LEASETIME"];
+                        if (strlen($DHCPleasetime) < 1) {
+                            // Fallback if empty string
+                            $DHCPleasetime = 24;
+                        }
+                    } else {
+                        $DHCPleasetime = 24;
+                    }
+                    if (isset($setupVars["DHCP_IPv6"])) {
+                        $DHCPIPv6 = $setupVars["DHCP_IPv6"];
+                    } else {
+                        $DHCPIPv6 = false;
+                    }
+                    if (isset($setupVars["DHCP_rapid_commit"])) {
+                        $DHCP_rapid_commit = $setupVars["DHCP_rapid_commit"];
+                    } else {
+                        $DHCP_rapid_commit = false;
+                    }
+
+                } else {
+                    $DHCP = false;
+                    // Try to guess initial settings
+                    if ($piHoleIPv4 !== "unknown") {
+                        $DHCPdomain = explode(".", $piHoleIPv4);
+                        $DHCPstart = $DHCPdomain[0] . "." . $DHCPdomain[1] . "." . $DHCPdomain[2] . ".201";
+                        $DHCPend = $DHCPdomain[0] . "." . $DHCPdomain[1] . "." . $DHCPdomain[2] . ".251";
+                        $DHCProuter = $DHCPdomain[0] . "." . $DHCPdomain[1] . "." . $DHCPdomain[2] . ".1";
+                    } else {
+                        $DHCPstart = "";
+                        $DHCPend = "";
+                        $DHCProuter = "";
+                    }
+                    $DHCPleasetime = 24;
+                    $DHCPIPv6 = false;
+                    $DHCP_rapid_commit = false;*/
+                }
+                /*if (isset($setupVars["PIHOLE_DOMAIN"])) {
+                    $piHoleDomain = $setupVars["PIHOLE_DOMAIN"];
+                } else {
+                    $piHoleDomain = "lan";
+                }*/
+                ?>
+                <div id="lancache" class="tab-pane fade <?php if($tab === "lancache"){ ?> in active<?php } ?>">
+                    <form role="form" method="post">
+                        <div class="row">
+                            <!-- Lancache settings box -->
+                            <div class="col-md-6">
+                                <div class="box box-warning">
+                                    <div class="box-header with-border">
+                                        <h3 class="box-title">Lancache settings</h3>
+                                    </div>
+                                    <div class="box-body">
+                                        <div class="row">
+                                            <div class="col-md-12">
+                                                <div class="form-group">
+                                                    <div class="checkbox">
+                                                        <label><input type="checkbox" name="active" id="LANCchk"
+                                                            <?php if ($LANC) {?>checked<?php }
+                                                            ?>>Lancache enabled</label>
+                                                    </div>
+                                                </div>
+                                            </div>
+                                        </div>
+                                        <div class="row">
+                                            <div class="col-xs-12">
+                                                <label>TODO</label>
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                            <!-- Lancache service settings box -->
+                            <div class="col-md-6">
+                                <div class="box box-warning">
+                                    <div class="box-header with-border">
+                                        <h3 class="box-title">Lancache services</h3>
+                                    </div>
+                                    <div class="box-body">
+                                        <div class="row">
+                                            <div class="col-md-12">
+                                                <table class="table table-bordered">
+                                                    <tr>
+                                                        <th>Name</th>
+                                                        <th>Description</th>
+                                                        <th colspan="2">Enabled</th>
+                                                    </tr>
+                                                    <?php foreach ($LCServices as $key => $value) { ?>
+                                                        <tr>
+                                                            <td><?php echo $key; ?></td>
+                                                            <td><?php echo $value['description']; ?></td>
+                                                            <td title="<?php echo $key; ?>">
+                                                                <input type="checkbox" name="LANC_<?php echo $key; ?>" value="true"
+                                                                           <?php if (isset($value["enabled"]) && $value["enabled"] == true){ ?>checked<?php } ?>>
+                                                            </td>
+                                                        </tr>
+                                                    <?php } ?>
+                                                </table>
+
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                    </form>
+				</div>
+				<!-- ######################################################### API and Web ######################################################### -->
                 <?php
                 // CPU temperature unit
                 if (isset($setupVars["TEMPERATUREUNIT"])) {
